src/popmanager.cpp

```diff
@@ -84,49 +82,41 @@ bool PopManager::addVTB(const VTB& vtb,
     if (!btc_->acceptBlock(block, state, &index)) {
       return state.addStackFunction("addVTB");
     }
-    stateChange.putBtcBlock(index);
   }
 
   // secondly, add VBK endorsements in BTC
   btce_->put(vtb);
-  stateChange.putBtcEndorsement(vtb);
 
   // thirdly, add vbk context blocks
   for (const auto& block : vtb.context) {
     BlockIndex<VbkBlock> index;
     if (!vbk_->acceptBlock(block, state, &index)) {
       return state.addStackFunction("addVTB");
     }
-    stateChange.putVbkBlock(index);
   }
 
   return true;
 }
 
-void PopManager::removeVTB(const VTB& vtb, StateChange& stateChange) noexcept {
+void PopManager::removeVTB(const VTB& vtb, StateChange&) noexcept {
   // remove VBK context in reverse order
   auto& vbkctx = vtb.context;
-  std::for_each(
-      vbkctx.rbegin(), vbkctx.rend(), [this, &stateChange](const VbkBlock& b) {
-        vbk_->invalidateBlockByHash(b.getHash());
-        stateChange.removeVbkBlock(b.getHash());
-      });
+  std::for_each(vbkctx.rbegin(), vbkctx.rend(), [this](const VbkBlock& b) {
+    vbk_->invalidateBlockByHash(b.getHash());
+  });
 
   // remove endorsement
   btce_->remove(vtb);
-  stateChange.removeBtcEndorsement(vtb);
 
   // remove BTC context in reverse order
   auto& btcctx = vtb.transaction.blockOfProofContext;
-  std::for_each(
-      btcctx.rbegin(), btcctx.rend(), [this, &stateChange](const BtcBlock& b) {
-        btc_->invalidateBlockByHash(b.getHash());
-        stateChange.removeBtcBlock(b.getHash());
-      });
+  std::for_each(btcctx.rbegin(), btcctx.rend(), [this](const BtcBlock& b) {
+    btc_->invalidateBlockByHash(b.getHash());
+  });
 }
 
 bool PopManager::addAltProof(const AltProof& payloads,
-                             StateChange& stateChange,
+                             StateChange&,
```

Comment:
How about deleting this?

Reply:
It is already deleted in the feature/alt-tree branch